### ISSUE TITLE
Provide ansible_distribution when building inventory 

### ISSUE THREAD

**Issue by HOSTED-POWER** [UpCloudLtd/upcloud-ansible-collection#7]
Hi,

We would need to know beforehand if a host if Windows, or Debain or CentOS, this way we can put them in appropriate groups.

For sure windows needs to be divided from the other hosts since other connection parameters are needed when using Ansible.

It would be great if more information of the server was specified.

Worst case just having the initial template name would be a great step forward ;)

Atm this is quite blocking to get 100% automation...

**Comment by ajmyyra** [comment]
Hi there!

We should be able to make this grouping available soon. Currently it's not yet available in our API (only in control panel) and you'd need to set this by yourself with our server labels (and then filtering with those), but as it will be made available with other changes, we can start supporting grouping these. I don't yet know the time when this will be available, but we should be getting back to this right after it's available.

**Comment by peter-soderkvist** [comment]
This is a rather old issue. Have this changed in your API so that it could be implemented? If I query the endpoint `https://api.upcloud.com/1.3/server/{uuid}` the OS info is in the storage labels.

```json
        {
          "address": "virtio:0",
          "boot_disk": "0",
          "labels": [
            {
              "key": "_os_architecture",
              "value": "64"
            },
            {
              "key": "_os_brand_name",
              "value": "Rocky"
            },
            {
              "key": "_os_brand_version",
              "value": "8.4"
            },
            {
              "key": "_os_main_category",
              "value": "Linux"
            },
            {
              "key": "_os_type",
              "value": "centos"
            },
            {
              "key": "_template_uuid",
              "value": "01000000-0000-4000-8000-000150010100"
            }
          ]
        }
```

**Comment by kangasta** [comment]
Yes, the storage labels feature has been added since this issue was created. These labels are available for storages in new servers.

If I recall right, `ansible_distribution` is automatically set when gathering facts, so not sure if we should alter the value from inventory plugin. Nevertheless, we could check that these storage labels are available in host data 🤔

**Comment by peter-soderkvist** [comment]
Yes, it's true that `ansible_distribution` is set when you set `gather_facts: true`. The issue with this is that you cannot run pre-tasks based on the OS to fetch connection vars etc, so it would be nice if the plugin could provide it even without gathering facts. We've worked our way around this by auto-generating our own inventory files, and it would be pretty sweet to use your plugin instead.
